### PR TITLE
[indexer grpc] Compression 

### PR DESCRIPTION
**Pull request by larry-aptos**

### CODE REVIEW

File: ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs [hunk 1]
```diff
@@ -221,62 +225,33 @@ async fn process_transactions_from_node_response(
                 SERVICE_TYPE,
                 IndexerGrpcStep::CacheWorkerReceivedTxns,
                 Some(start_version as i64),
-                Some(last_transaction.version as i64),
+                Some(last_transaction_version as i64),
                 first_transaction_pb_timestamp.as_ref(),
                 last_transaction_pb_timestamp.as_ref(),
                 Some(batch_start_time.elapsed().as_secs_f64()),
                 Some(size_in_bytes),
-                Some((last_transaction.version - first_transaction.version + 1) as i64),
```
Comment (larry-aptos) [inline ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs:229]: add the grpc logging inside storage_format.

File: ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs [hunk 2]
```diff
@@ -479,108 +452,40 @@ async fn data_fetch(
...
-            let end_version_timestamp = {
-                let decoded_transaction =
-                    base64::decode(transactions.last().unwrap()).expect("Failed to decode base64.");
-                let transaction =
-                    Transaction::decode(&*decoded_transaction).expect("Failed to decode protobuf.");
-                transaction.timestamp
-            };
-
-            log_grpc_step(
-                SERVICE_TYPE,
```
Comment (larry-aptos) [inline ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs:514]: add log_grpc_step

File: ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs [hunk 3]
```diff
@@ -479,108 +452,40 @@ async fn data_fetch(
...
-                    let end_version_timestamp = {
-                        let decoded_transaction = base64::decode(transactions.last().unwrap())
-                            .expect("Failed to decode base64.");
-                        let transaction = Transaction::decode(&*decoded_transaction)
-                            .expect("Failed to decode protobuf.");
-                        transaction.timestamp
-                    };
-
-                    log_grpc_step(
-                        SERVICE_TYPE,
```
Comment (larry-aptos) [inline ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs:558]: add log_grpc_step

File: ecosystem/indexer-grpc/indexer-grpc-utils/src/file_store_operator/local.rs [hunk 4]
```diff
@@ -141,43 +158,43 @@ impl FileStoreOperator for LocalFileStoreOperator {
...
             "Starting version has to be a multiple of BLOB_STORAGE_SIZE."
         );
         anyhow::ensure!(
-            batch_size % BLOB_STORAGE_SIZE == 0,
+            batch_size % FILE_ENTRY_TRANSACTION_COUNT as usize == 0,
             "The number of transactions to upload has to be multiplier of BLOB_STORAGE_SIZE."
         );
         let mut tasks = vec![];
 
-        // create files directory
```
Comment (larry-aptos) [inline ecosystem/indexer-grpc/indexer-grpc-utils/src/file_store_operator/local.rs:158]: Fix the folder creation

File: ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs (hunk 1 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs:229]: done.

File: ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs [hunk 5]
```diff
@@ -345,93 +359,96 @@ async fn process_streaming_response(
...
+                        let start_version = status.start_version;
+                        let num_of_transactions = status
+                            .end_version
+                            .expect("TransactionsFromNodeResponse status end_version is None")
+                            + 1
+                            - start_version;
+                        tx.send(CacheTaskItem::UpdateCacheLatestVersion((
+                            num_of_transactions,
+                            start_version,
+                        )))
```
Comment (rtso) [inline ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs:378]: why do we need a separate item for this? this makes it more complicated than it needs to be. why not spin up 10 threads, process batch of size X, check for gaps, then update cache latest version. similar to how it's done in the filestore worker now.

**Comment by larry-aptos** [comment]
replaced by lz4

**Pull request closed without merging**